### PR TITLE
FIx issue #317

### PR DESCRIPTION
**Pull request by ngocnhan2003**

### CODE REVIEW

**Comment by ArslanAmeer** [comment]
@anmol098 

Can you look into this PR and merge asap.

**Comment by anmol098** [comment]
The PR seems to have conflicts.

**Comment by ArslanAmeer** [comment]
@anmol098 .
Conflicts resolved. Can you take look into it

**Comment by ArslanAmeer** [comment]
@ngocnhan2003 is it you reverting your own commits?

**Comment by ngocnhan2003** [comment]
> @ngocnhan2003 is it you reverting your own commits?

Yes, both accounts are mine. The reverting commit is correct.

**Comment by ArslanAmeer** [comment]
@anmol098 PR is ready to be merged

File: main.py [hunk 1]
```diff
@@ -563,9 +565,12 @@ def generate_new_readme(stats: str, readme: str):
         rdmd = decode_readme(contents.content)
         new_readme = generate_new_readme(stats=waka_stats, readme=rdmd)
         if commit_by_me.lower() in truthy:
-            committer = InputGitAuthor(username, email)
+            committer = InputGitAuthor(username or commit_username, email or commit_email)
```
Comment (oHTGo) [inline main.py:568]: Excuse me
If commit_username is empty, it will have exception

File: main.py [hunk 2]
```diff
@@ -563,9 +565,12 @@ def generate_new_readme(stats: str, readme: str):
         rdmd = decode_readme(contents.content)
         new_readme = generate_new_readme(stats=waka_stats, readme=rdmd)
         if commit_by_me.lower() in truthy:
-            committer = InputGitAuthor(username, email)
+            committer = InputGitAuthor(username or commit_username, email or commit_email)
         else:
-            committer = InputGitAuthor('readme-bot', '41898282+github-actions[bot]@users.noreply.github.com')
+            committer = InputGitAuthor(
+                commit_username or 'readme-bot',
```
Comment (oHTGo) [inline main.py:571]: Why do you set the default value in action.yml?

**Comment by ArslanAmeer** [comment]
@ngocnhan2003

File: main.py (hunk 2 above)
Comment (ngocnhan2003) [reply main.py:571]: 41898282 is GitHub action bot ID, if the user does not set commit_by_me so we can define a bot for commiting.

Comment (ngocnhan2003) [reply main.py:571]: and GitHub API now did not return email so I add field commit_email that we can provide our email.

**Comment by ArslanAmeer** [comment]
@anmol098 This PR is ready

**Pull request merged**